service/controller/machinepool/handler/migration/resource.go

```diff
@@ -0,0 +1,171 @@
+package migration
+
+import (
+	"context"
+	"encoding/json"
+	"reflect"
+
+	oldcapiexpv1alpha3 "github.com/giantswarm/apiextensions/v6/pkg/apis/capiexp/v1alpha3"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/operatorkit/v7/pkg/controller/context/reconciliationcanceledcontext"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	capz "sigs.k8s.io/cluster-api-provider-azure/api/v1beta1"
+	capi "sigs.k8s.io/cluster-api/api/v1beta1"
+	capiexp "sigs.k8s.io/cluster-api/exp/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/giantswarm/azure-operator/v5/service/controller/key"
+)
+
+const (
+	// Name is the identifier of the resource.
+	Name = "migration"
+)
+
+var (
+	DesiredCAPIGroupVersion = capi.GroupVersion.String()
+	DesiredCAPZGroupVersion = capz.GroupVersion.String()
+)
+
+type Config struct {
+	CtrlClient client.Client
+	Logger     micrologger.Logger
+}
+
+type Resource struct {
+	ctrlClient client.Client
+	logger     micrologger.Logger
+}
+
+func New(c Config) (*Resource, error) {
+	if c.CtrlClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.CtrlClient must not be empty", c)
+	}
+	if c.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", c)
+	}
+
+	r := &Resource{
+		ctrlClient: c.CtrlClient,
+		logger:     c.Logger,
+	}
+
+	return r, nil
+}
+
+func (r Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
```

Comment:
Handler that finishes MachinePool migration that got started by old experimental MachinePool handler:
- it updates Status of the new MachinePool by copying it from the old MachinePool
- it deletes old experimental MachinePool
- it cancels new MachinePool reconciliation if the migration is not complete